src/Common/src/TypeSystem/IL/ILProvider.cs

```diff
@@ -31,10 +31,18 @@ private MethodIL TryGetIntrinsicMethodIL(MethodDesc method)
             if (owningType == null)
                 return null;
 
-            if (method.Name == "UncheckedCast" && owningType.Name == "RuntimeHelpers" && owningType.Namespace == "System.Runtime.CompilerServices")
+            string methodName = method.Name;
+
```

Comment:
Should we also check the assembly for the type to ensure we are not working with types defined in user code?